### PR TITLE
Feature/bound

### LINKED ISSUE #609: namelist for ww3_bound
**Describe the solution you'd like**
namelist feature for program ww3_bound

**Additional context**
can be tested with existing regtest ww3_tr1

### PR DESCRIPTION
**Pull request by mickaelaccensi**
# Pull Request Summary
namelist feature for program ww3_bound

## Description
namelist is implemented for program ww3_bound and validated with regtest ww3_tr1

### Issue(s) addressed
- fixes #609 

### Commit Message
namelist feature for program ww3_bound

### Check list  

- [X] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [X] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [X] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [X] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested?
only on ww3_tr1 with namelist option -N
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
ww3_tr1
* Have the matrix regression tests been run (if yes, please note HPC and compiler)?
only ww3_tr1 with namelist option on datarmor with compiler intel
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):  
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/8047487/matrixCompSummary.txt)
[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/8047488/matrixCompFull.txt)
* Please indicate the expected changes in the regression test output ([Note the known list of non-identical tests](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-master#4-look-at-results)).
no changes in the output. only difference is an additional log file with the namelist feature

### CODE REVIEW

File: model/src/w3nmlboundmd.F90 [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+!/ ------------------------------------------------------------------- /
+
+
+
+
+END MODULE W3NMLBOUNDMD
+
+!/ ------------------------------------------------------------------- /
+
+
```
Comment (JessicaMeixner-NOAA) [inline model/src/w3nmlboundmd.F90]: remove extra lines at the end

File: model/src/w3nmlboundmd.F90 [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+!
+!/ ------------------------------------------------------------------- /
+
+  ! module defaults
+  IMPLICIT NONE
+
+  PUBLIC
+
+  ! bound structure
+  TYPE NML_BOUND_T
```
Comment (JessicaMeixner-NOAA) [inline model/src/w3nmlboundmd.F90:27]: When we add in the doxygen, we'll need variable descriptions for all of these variables in the routines.  Maybe we should just add these now? (even if not in doxygen format, just give quick descriptions of the variables).

**Comment by JessicaMeixner-NOAA** [comment]
I'll start running the regtests.

**Comment by JessicaMeixner-NOAA** [comment]
@mickaelaccensi the output from the regtests: 

```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (18 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (5 files differ)
ww3_ufs1.1/./work_d                     (0 files differ)
ww3_ufs1.2/./work_b                     (0 files differ)
```


[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/8061849/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/8061851/matrixCompSummary.txt)
[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/8061852/matrixDiff.txt)

It would be nice to get the variable defined now so that we'll have that information when we add the doxygen documentation.   Alternatively, we can make an issue and assign it to you that you will add the doxygen with the variables defined.   Please let me know what your preference is.

**Comment by mickaelaccensi** [comment]
done !

**Comment by JessicaMeixner-NOAA** [comment]
> done !

THANK YOU!!!!

**Comment by JessicaMeixner-NOAA** [comment]
Just the expected non-identical cases: 
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (6 files differ)
ww3_ufs1.1/./work_d                     (0 files differ)
ww3_ufs1.2/./work_b                     (0 files differ)
ww3_ufs1.3/./work_a                     (1 files differ)
```

[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/8080127/matrixDiff.txt)
[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/8080128/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/8080130/matrixCompSummary.txt)

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Pull request merged**